graaljs/graaljs-gradle-webpack-guide/README.md

```diff
@@ -0,0 +1,187 @@
+
+1. **Build the application**:
+
+   ```sh
+   gradle build
+   ```
+
+2. **Run the application**:
+
+   ```sh
+   gradle run --args="https://www.graalvm.org/"
+   ```
+
+   After running the application, you should see the generated QR code printed as output.
+
+## Java Code Overview
+
+### `App.java`
+This is the main class that initializes the GraalVM context and executes the JavaScript QR code generation. The code integrates the `qrcode` NPM package by loading the JavaScript bundle (`bundle.mjs`) and calling the QR code generation function.
+
+```java
+try (Context context = Context.newBuilder("js")
+        .allowHostAccess(HostAccess.ALL)
+        .option("engine.WarnInterpreterOnly", "false")
+        .option("js.esm-eval-returns-exports", "true")
+        .option("js.unhandled-rejections", "throw")
+        .build()) {
+
+    // Load the JavaScript bundle
+    Source bundleSrc = Source.newBuilder("js", App.class.getResource("/bundle/bundle.mjs")).build();
+    Value exports = context.eval(bundleSrc);
+
+    // Set the input URL for QR code generation
+    String input = args.length > 0 ? args[0] : "https://www.graalvm.org/javascript/";
+
+    // Get QRCode object and generate QR code as a string
+    QRCode qrCode = exports.getMember("QRCode").as(QRCode.class);
+    Promise resultPromise = qrCode.toString(input);
+
+    resultPromise.then((result) -> {
+        System.out.println("Successfully generated QR code for \"" + input + "\".");
+        System.out.println(result.asString());
+    });
+}
+```
+
+### `QRCode.java`
+```java
+public interface QRCode {
+    Promise toString(String data);
+    Promise toDataURL(String data, Object options);
+}
+```
+
+### `Promise.java`
+```java
+public interface Promise {
+    Promise then(ValueConsumer onResolve);
+    Promise then(ValueConsumer onResolve, ValueConsumer onReject);
+}
+```
+
+## JavaScript Code Overview
+
+### `main.mjs`
+```js
+import 'fast-text-encoding';
+export * as QRCode from 'qrcode';
+```
+
+### Webpack Configuration
+
+#### `webpack.config.js`
+```js
+const path = require('path');
+const { EnvironmentPlugin } = require('webpack');
+
+const isProduction = process.env.NODE_ENV == 'production';
+
+const config = {
+    entry: './main.mjs',
+    output: {
+        path: process.env.BUILD_DIR
+            ? path.resolve(process.env.BUILD_DIR)
+            : path.resolve(__dirname, '../../build/classes/java/main/bundle'),
+        filename: 'bundle.mjs',
+        module: true,
+        library: {
+            type: 'module',
+        },
+        globalObject: 'globalThis'
+    },
+    experiments: {
+        outputModule: true
+    },
+    mode: isProduction ? 'production' : 'development',
+};
+
+module.exports = () => config;
+```
+
+## Gradle Configuration
+
+#### `build.gradle`
+This Gradle configuration sets up the Java build environment, including the necessary dependencies for GraalVM and Node.js.
+
+```gradle
+plugins {
```

Comment:
This appears to be a copy of the entire _build.gradle_ file, which is already included in the repo and can be examined by the user. Instead, I think it would make sense to only show what needs to be added, similar to how it's done in the Maven guide (e.g., [here](https://github.com/graalvm/graal-languages-demos/blob/0951909cb64a04262eaa6a27bc5800846e785019/graaljs/graaljs-maven-webpack-guide/README.md?plain=1#L38) and [here](https://github.com/graalvm/graal-languages-demos/blob/0951909cb64a04262eaa6a27bc5800846e785019/graaljs/graaljs-maven-webpack-guide/README.md?plain=1#L72)).